src/app/ReadClient.cpp

```diff
@@ -520,6 +520,15 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
 CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
 {
     CancelLivenessCheckTimer();
+    uint64_t timeoutMsec = mMaxIntervalCeilingSeconds;
```

Comment:
Quite apart from how this is or should be computed ... this variable is not used for anything, right?  How was this tested?

Also, there is some interesting unit confusion going on here with seconds vs msec?

Reply:
improved it by System::Clock , thanks